### PR TITLE
Release v0.3.1

### PR DESCRIPTION
**Pull request by astoycos**

### CODE REVIEW

**Comment by astoycos** [comment]
Still running into some issues deploying on Openshift

**Review by anfredette — DISMISSED** [review]
Otherwise, LGTM

File: RELEASE.md
[the remark this answers is not in the document]
Comment (astoycos) [reply RELEASE.md:1]: Will do Waiting for some more reviews.

File: bpfd-operator/ocp.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+apiVersion: v1
```
Comment (astoycos) [inline bpfd-operator/ocp.txt:1]: YUCK DELETE

File: docs/developer-guide/operator-quick-start.md [hunk 2]
```diff
@@ -1,14 +1,11 @@
 # Deploying the bpfd-operator
 
-The `bpfd-operator` repository exists in order to deploy and manage bpfd within a Kubernetes cluster.
-
-## Deploy
-
+The `bpfd-operator` repository exists in order to deploy and manage bpfd within a Kubernetes cluster.\
```
Comment (dave-tucker) [inline docs/developer-guide/operator-quick-start.md:3]: Trailing `\`

File: changelogs/CHANGELOG-v0.3.1.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+The v0.3.1 release is a patch release which includes some bug fixes and makes
+the ability for bpf enabled applications to receive their maps via bpfd's
+custom CSI plugin default.
+
+This enablement means that most applications can now access their maps **WITHOUT**
+being run as root.
+
+> **WARNING**
```
Comment (dave-tucker) [inline changelogs/CHANGELOG-v0.3.1.md:8]:
```suggestion
> [!WARNING]
```

Will make this prettier.

File: changelogs/CHANGELOG-v0.3.1.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+The CSI feature still requires a privileged application on distributions which
+enable seLinux by default (i.e openshift).  Therefore we've shipped a set of
```
Comment (Billy99) [inline changelogs/CHANGELOG-v0.3.1.md:10]:
```suggestion
enable SELinux by default (i.e Red Hat Openshift).  Therefore we've shipped a set of
```

File: changelogs/CHANGELOG-v0.3.1.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+deployment configs specifically for openshift in this release, see the additional
+`go-<example>-counter-install-ocp-v0.3.1.yaml`` artifacts included in the release
+payload. Stay tuned to [#829](https://github.com/bpfd-dev/bpfd/issues/829) for
+updates.
+
+The new yaml syntax that should be used by bpf enabled applications resembles
```
Comment (Billy99) [inline changelogs/CHANGELOG-v0.3.1.md:16]:
```suggestion
The new yaml syntax that should be used by BPF enabled applications resembles
```

File: changelogs/CHANGELOG-v0.3.1.md (hunk 3 above)
Comment (dave-tucker) [reply changelogs/CHANGELOG-v0.3.1.md:8]: The older syntax was just deprecated earlier this month: https://github.com/orgs/community/discussions/16925

File: changelogs/CHANGELOG-v0.3.1.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        - name: bpf-maps
+          csi:
+            driver: csi.bpfd.dev
+            volumeAttributes:
+              csi.bpfd.dev/program: <*Program_Name>
+              csi.bpfd.dev/maps: <BPF Map Names>
+```
+
+Additionally, this release removes all dependencies involved with deploying bpfd
+with TLS, meaning cert-manager deps are completely removed from the operator
```
Comment (Billy99) [inline changelogs/CHANGELOG-v0.3.1.md:44]:
```suggestion
with TLS, meaning cert-manager dependencies are completely removed from the operator,
```

File: changelogs/CHANGELOG-v0.3.1.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+therefore simplifying the deployment considerably.
+
```
Comment (Billy99) [inline changelogs/CHANGELOG-v0.3.1.md:46]: Do we need to highlight that the bpfd user/group was removed?

File: changelogs/CHANGELOG-v0.3.1.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+* bpfd: Use tc dispatcher from container image by @dave-tucker in https://github.com/bpfd-dev/bpfd/pull/817
+* bpfd: Unify the "run as root" and "run as bpfd user" codepaths by @Billy99 in https://github.com/bpfd-dev/bpfd/pull/777
+* bpfd, bpfctl, operator: Remove support for TCP/TLS by @dave-tucker in https://github.com/bpfd-dev/bpfd/pull/819
+* bpfd-operator: Make the CSI deployment default for bpfd-operator by @Billy99 in https://github.com/bpfd-dev/bpfd/pull/811
+* ci: Add YAML formatter by @dave-tucker in https://github.com/bpfd-dev/bpfd/pull/802
+* Fix some panics + add testing and fix for map sharing  by @astoycos in https://github.com/bpfd-dev/bpfd/pull/820
+* bpfd: mount default bpffs on kind by @astoycos in https://github.com/bpfd-dev/bpfd/pull/823
+* bpfd: Remove unused file by @dave-tucker in https://github.com/bpfd-dev/bpfd/pull/824
+* Document valid kernel versions by @Billy99 in https://github.com/bpfd-dev/bpfd/pull/827
+
```
Comment (Billy99) [inline changelogs/CHANGELOG-v0.3.1.md:67]: * Update documentation on new YAML Linter by @Billy99 in https://github.com/bpfd-dev/bpfd/pull/830

**Review by Billy99 — COMMENTED** [review]
Just some nits

File: changelogs/CHANGELOG-v0.3.1.md (hunk 7 above)
Comment (astoycos) [reply changelogs/CHANGELOG-v0.3.1.md:46]: I'm not sure.... I guess it is sort of a user facing change if anyone was running bpfd with systemd previously? 

So yeah we should

File: changelogs/CHANGELOG-v0.3.1.md (hunk 6 above)
Comment (anfredette) [reply changelogs/CHANGELOG-v0.3.1.md:44]: :s/meaning/which means that/

File: changelogs/CHANGELOG-v0.3.1.md (hunk 3 above)
Comment (astoycos) [reply changelogs/CHANGELOG-v0.3.1.md:8]: Sweet thanks I didn't know that!

File: changelogs/CHANGELOG-v0.3.1.md (hunk 7 above)
Comment (astoycos) [reply changelogs/CHANGELOG-v0.3.1.md:46]: Done

**Pull request merged**